packages/shell-api/src/collection.ts

```diff
@@ -2074,12 +2080,17 @@ export default class Collection extends ShellApiWithMongoClass {
   @returnsPromise
   @apiVersions([])
   // TODO(MONGOSH-1471): use SearchIndexDescription once available
-  async createSearchIndex(description: any): Promise<string> {
-    this._emitCollectionApiCall('createSearchIndex', { description });
+  async createSearchIndex(indexName?: string|Document, description?: Document): Promise<string> {
+    if (typeof indexName === 'object' && indexName !== null) {
+      description = indexName;
+      indexName = undefined;
+    }
+
+    this._emitCollectionApiCall('createSearchIndex', { indexName, description });
     const results = await this._mongo._serviceProvider.createSearchIndexes(
       this._database._name,
       this._name,
-      [description]
+      [{ name: (indexName as string|undefined) ?? 'default', description: { ...description } }]
```

Comment:
The doc says that with no name it should go with `'default'`. Not clear if we should do it, the driver should do it or the server should do it. I can't see the driver doing it and since there's no way for me to test what the server does I'm just filling in the default here.